File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/dto/cafe/CafeSearchResponseDto.java
package io.beaniejoy.dongnecafe.domain.cafe.dto.cafe;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CafeSearchResponseDto {

    private UUID cafeId;

    private String name;

    private String address;

    private Double totalRate;

    private List<CafeImageResponseDto> imageList;
}

File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/dto/cafe/CafeSearchResponseDto.kt
package io.beaniejoy.dongnecafe.domain.cafe.dto.cafe

import io.beaniejoy.dongnecafe.domain.cafe.domain.Cafe

data class CafeSearchResponseDto(
    val id: Long = 0L,
    val name: String? = null,
    val address: String? = null,
    val totalRate: Double? = null,
    val imageList: List<CafeImageResponseDto> = emptyList()
) {
    companion object {
        fun of(cafe: Cafe): CafeSearchResponseDto {
            return CafeSearchResponseDto(
                id = cafe.id,
                name = cafe.name,
                address = cafe.address,
                totalRate = cafe.totalRate,
                imageList = cafe.cafeImageList.map { CafeImageResponseDto.of(it) }
            )
        }
    }
}